src/components/shared/ToggleView.tsx

```diff
@@ -0,0 +1,56 @@
+import React from 'react';
+import classnames from 'classnames';
+import { inject, observer } from 'mobx-react';
+import { SearchStore } from '~/stores/searchStore';
+import * as s from './toggleView.scss';
+
+export interface ToggleItem {
+    icon?: JSX.Element;
+    text: string;
+    isActive: boolean;
+    onClick(): void;
+}
+
+interface IEntityTypeToggleProps {
+    searchStore?: SearchStore;
+    toggles: ToggleItem[];
+}
+
+@inject('searchStore')
+@observer
+class ToggleView extends React.Component<IEntityTypeToggleProps> {
+
+    render() {
+        return (
+            <div className={s.toggleView}>
+                <div className={s.buttonContainer}>
```

Comment:
Would suggest this to use React syntax.
< ToggleView>
<ToggleButton
 icon=
 text=
 isActive=
 onClick=
/>
<ToggleButton...
</ ToggleView>